engine/table/src/main/java/io/deephaven/engine/table/impl/SourcePartitionedTable.java

```diff
@@ -219,22 +241,66 @@ private void processPendingLocations(final boolean notifyListeners) {
-                result.notifyListeners(new TableUpdateImpl(added,
-                        RowSetFactory.empty(), RowSetFactory.empty(), RowSetShiftData.EMPTY, ModifiedColumnSet.EMPTY));
-            } else {
-                added.close();
-            }
+            final RowSet added = sortAndAddLocations(readyLocationStates.stream()
+                    .map(PendingLocationState::release));
             readyLocationStates.clearFast();
+            return added;
+        }
+
+        private RowSet processRemovals(final TableLocationSubscriptionBuffer.LocationUpdate locationUpdate) {
+            final Set<ImmutableTableLocationKey> relevantRemovedLocations = locationUpdate.getPendingRemovedLocations()
+                    .stream()
+                    .map(TableLocation::getKey)
+                    .filter(locationKeyMatcher)
+                    .collect(Collectors.toSet());
+
+            if (relevantRemovedLocations.isEmpty()) {
+                return RowSetFactory.empty();
+            }
+
+            final RowSetBuilderSequential deleteBuilder = RowSetFactory.builderSequential();
+
+            // We don't have a map of location to row key, so we have to iterate them. If we decide this is too slow, we
+            // could add a TObjectIntMap
+            // as we process pending added locations and then we can just make an index of rows to remove by looking up
+            // in that map.
+            try (final ChunkSource.GetContext locContext = resultTableLocationKeys.makeGetContext(CHUNK_CAPACITY);
+                    final ChunkSource.GetContext tableContext = resultLocationTables.makeGetContext(CHUNK_CAPACITY);
+                    final RowSequence.Iterator it = resultRows.getRowSequenceIterator()) {
+
+                while (it.hasMore()) {
+                    final RowSequence subSeq = it.getNextRowSequenceWithLength(CHUNK_CAPACITY);
+                    final LongChunk<OrderedRowKeys> keyChunk = subSeq.asRowKeyChunk();
+                    final ObjectChunk<Table, ? extends Values> tableChunk =
+                            resultLocationTables.getChunk(tableContext, subSeq)
+                                    .asObjectChunk();
+                    final ObjectChunk<ImmutableTableLocationKey, ? extends Values> removedKeys = resultTableLocationKeys
+                            .getChunk(locContext,
+                                    subSeq)
+                            .asObjectChunk();
+
+                    for (int chunkPos = 0; chunkPos < keyChunk.size(); chunkPos++) {
+                        if (relevantRemovedLocations.contains(removedKeys.get(chunkPos))) {
+                            deleteBuilder.appendKey(keyChunk.get(chunkPos));
+
+                            final Table deletedTable = tableChunk.get(chunkPos);
+                            result.unmanage(deletedTable);
+                            deletedTable.close();
+                        }
+                    }
+                }
+            }
```

Comment:
This code would be prettier (although maybe a little less efficient) if you wrote it with column iterators. Actually, now that I see you using key chunks, I suspect it might just be better the other way.

Regardless of whether you take my suggestion, `resultTableLocationKeys` should be changed to hold `ImmutableTableLocationKey`.

```suggestion
            // @formatter:off
            try (final CloseableIterator<ImmutableTableLocationKey> keysIterator =
                         ChunkedObjectColumnIterator.make(resultTableLocationKeys, resultRows);
                 final CloseableIterator<Table> constituentsIterator =
                         ChunkedObjectColumnIterator.make(resultLocationTables, resultRows);
                 final RowSet.Iterator rowsIterator = resultRows.iterator()) {
                // @formatter:on

                while (keysIterator.hasNext()) {
                    final TableLocationKey key = keysIterator.next();
                    final Table constituent = constituentsIterator.next();
                    final long rowKey = rowsIterator.nextLong();
                    if (relevantRemovedLocations.contains(key)) {
                        deleteBuilder.appendKey(rowKey);
                        result.unmanage(constituent);
                        constituent.close();
                    }
                }
            }
```